### PR TITLE
feature/#1/Update&Delete endpoints

### PR DESCRIPTION
**Pull request by politopolito**
# Pull Request

## Type of Change

- [ X] New feature (non-breaking change which adds functionality)

## Branch information

* Resolves: #(issue)
* **Description** Add updateById and delete using put and delete methods, adding modification on files to support it.

## Checklist

* [ ] Adds new tests
* [ ] New and existing test suites succeed
* [ X ] Linter succeeds
* [ ] Updates corresponding documentation

### CODE REVIEW

File: src/services/PhotoService.ts [hunk 1]
```diff
@@ -17,4 +18,15 @@ export default class PhotoService implements IService<Photo>{
     console.log(payload);
     return PhotoService.getRepository().save({ ...payload });
   }
+
+  public async updateById(id:number, payload: PhotoUpdateValidator): Promise<Photo> {
+    console.log(payload);
```
Comment (herrera-ignacio) [inline src/services/PhotoService.ts]: Remove debugging console.log

File: src/validators/Photo/PhotoUpdateValidator.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { IsString } from "class-validator";
+import { Transform } from "class-transformer";
+import PhotoValidator from "./PhotoValidator";
+
+export default class PhotoUpdateValidator extends PhotoValidator {
```
Comment (herrera-ignacio) [inline src/validators/Photo/PhotoUpdateValidator.ts:5]: This is good, but we should also consider other updatable fields such as `isPublic`.

File: src/routes/PhotoRouter.ts [hunk 3]
```diff
@@ -20,6 +21,8 @@ class PhotoRouter implements IRouter {
   initializeRoutes() {
     this.router.get(`${this.path}/:id(\\d+)`, this.photoController.getById);
     this.router.post(this.path, myValidator(PhotoCreateValidator, true), this.photoController.create);
+    this.router.put(`${this.path}/:id(\\d+)`, myValidator(PhotoUpdateValidator, true), this.photoController.updateById);
```
Comment (herrera-ignacio) [inline src/routes/PhotoRouter.ts]: Consider using `PATCH` in this case. We should use `PUT` only when we expect to receive a complete entity (i.e., with all its required fields) that completely replaces what we already have. We should use `PATCH` when we want to apply a partial update to an existing entity. 

```suggestion
    this.router.patch(`${this.path}/:id(\\d+)`, myValidator(PhotoUpdateValidator, true), this.photoController.updateById);
```

> TL;DR [An interesting discussion about this in SO](https://stackoverflow.com/questions/28459418/use-of-put-vs-patch-methods-in-rest-api-real-life-scenarios).

File: src/services/PhotoService.ts [hunk 4]
```diff
@@ -17,4 +18,15 @@ export default class PhotoService implements IService<Photo>{
...
+    const data = await PhotoService.getRepository().findById(id);
```
Comment (herrera-ignacio) [inline src/services/PhotoService.ts]: What would happen if we try to update a photo that doesn't exist 😄 ?

File: src/controllers/PhotoController.ts [hunk 5]
```diff
@@ -16,6 +16,19 @@ class PhotoController implements IController {
...
+    const photoId = Number(req.params.id);
+    const photoPayload = req.body;
+    const photo = await this.photoService.updateById(photoId, photoPayload);
+    res.status(201).json({ data: photo });
+  };
+
+  public deleteById: IRequestHandler = async (req, res) => {
+    const photoId = Number(req.params.id);
+    const photo = await this.photoService.deleteById(photoId);
+    res.status(201).json({ data: photo });
```
Comment (herrera-ignacio) [inline src/controllers/PhotoController.ts]: 201 Status code should be used for created only, in this case we should use 204.

File: src/controllers/PhotoController.ts [hunk 6]
```diff
@@ -16,6 +16,19 @@ class PhotoController implements IController {
...
+  public updateById: IRequestHandler = async (req, res) => {
+    const photoId = Number(req.params.id);
+    const photoPayload = req.body;
+    const photo = await this.photoService.updateById(photoId, photoPayload);
+    res.status(201).json({ data: photo });
+  };
+
+  public deleteById: IRequestHandler = async (req, res) => {
+    const photoId = Number(req.params.id);
+    const photo = await this.photoService.deleteById(photoId);
```
Comment (herrera-ignacio) [inline src/controllers/PhotoController.ts]: But, does it actually return a photo?

File: src/controllers/PhotoController.ts [hunk 7]
```diff
@@ -16,6 +16,19 @@ class PhotoController implements IController {
     const photo = await this.photoService.create(photoPayload);
     res.status(201).json({ data: photo });
   };
+
+  public updateById: IRequestHandler = async (req, res) => {
+    const photoId = Number(req.params.id);
+    const photoPayload = req.body;
+    const photo = await this.photoService.updateById(photoId, photoPayload);
+    res.status(201).json({ data: photo });
```
Comment (herrera-ignacio) [inline src/controllers/PhotoController.ts]: 201 should be used for created only, in this case we should use 204.

**Review by herrera-ignacio — CHANGES_REQUESTED** [review]
